lib/gems/pending/util/extensions/miq-benchmark.rb

```diff
@@ -29,6 +29,10 @@ def self.realtime_block(key, &block)
     begin
       ret = realtime_store(hash, key, &block)
       return ret, hash
+    rescue
+      # Don't let timings be lost on exception when there is nobody else to pick them up.
+      $log.info("Exception in realtime_block #{key.inspect} - Timings: #{hash.inspect}") if outermost
```

Comment:
I'm clueless about how all the `$log`,  `_log` thingies come to be (storks?), please advise...

The OOP answer is not use Benchmark class methods but an object intialized with a logger.  meh — much hassle, and pushes similar problem onto the users that would want a global instance...

- `realtime_block` could take a logger option, but again too much hassle to pass it everywhere, especially for an edge case.
- could have some accessor `Benchmark.realtime_block_logger = ...` to be initialized once in apps that have logging.  (need advice where to set in MIQ?)
- just default to logging to stderr, might be good enough for a rare edge case?

I also wanted to propose a convenience `realtime_log_once` function that logs timing IFF it's the outermost, and only returns the result, so callers don't need to discard `result, _ = ...`.
Same considerations, except it's standard flow, not an edge case.